### PR TITLE
 fix: Remove unrelated output in main_binary 

### LINKED ISSUE #45: stderr().is() fails due to cargo output
Hey, during my first attempts of writing integration tests I've stumbled upon an issue when the program is supposed to exit with an error output.

My test looks like this:

```rust
    #[test]
    fn called_without_args() {
        assert_cli::Assert::main_binary()
            .fails_with(::libc::EINVAL)
            .and()
            .stderr().is(MISSING_ARGUMENTS_OUTPUT)
            .unwrap();
    }
```

```diff
test integration::called_without_args ... FAILED

failures:

---- integration::called_without_args stdout ----
        thread 'integration::called_without_args' panicked at 'CLI assertion failed: `cargo run --` StdErr mismatch: diff:

+Compiling invsearch v0.1.0 (file:///.../invsearch.rs)
    Finished dev [unoptimized + debuginfo] target(s) in 0.62 secs
     Running `target/debug/invsearch`
 No arguments provided. Please provide files to parse.
', /home/steveej/.cargo/registry/src/github.com-1ecc6299db9ec823/assert_cli-0.5.3/src/assert.rs:319:12
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```

The issue is that the error outputs of cargo and the integration test are mixed and the test fails because it's not an exact match.
I know that I could just use `.contains(..)` instead of `.is(..)`, but IMHO that's only a workaround.

Please let me know if I'm doing something unexpected, as this is my very first attempt of writing (integration) tests in Rust.

### PR DESCRIPTION
**Pull request by epage**
To help in validating this, tests have been added for `main_binary` and `cargo_binary`. 

Fixed #45

### CODE REVIEW

**Comment by epage** [comment]
I'm a bit baffled at the failure.  stable and nightly succeeded.  beta failed with
```
---- readme_sect_example_line_41 stdout ----
	thread 'readme_sect_example_line_41' panicked at 'failed to read dependencies: Error(Toml(Error { inner: ErrorInner { kind: Custom, line: None, col: 0, message: "missing field `root`", key: [] } }), State { next_error: None, backtrace: None })', /checkout/src/libcore/result.rs:906:4
```

File: Cargo.toml [hunk 1]
```diff
@@ -12,6 +12,9 @@ categories = ["development-tools::testing"]
 keywords = ["cli", "testing", "assert"]
 build = "build.rs"
 
+[[bin]]
+name = "assert_fixture"
```
[the remark this answers is not in the document]
Comment (epage) [reply Cargo.toml]: > This binary is built only when building assert_cli itself, right? I.e., not when using it as a dependency. 

I think I've heard depending on something doesn't pull in the bins.

>  Is there any chance people might cargo install this?

If they did, it wouldn't do them much good.

>  We could add a internal-tests feature and specify required-features for the bin. Or, can we add bins depending on cfg(test) in some other way?

The problem with relying on features is `main_binary` can't enable the feature.

**Comment by epage** [comment]
> I'm a bit baffled at the failure. stable and nightly succeeded. beta failed with

> Is this a known bug with either rust beta or skeptic?

I've run into this with liquid also (beta, also depends on skeptic).  At least on slack, no one else has run into this.  I've not created a skeptic issue yet or checked on IRC.

**Comment by epage** [comment]
Reported this to Skeptic
https://github.com/budziq/rust-skeptic/issues/65

**Comment by epage** [comment]
Since there is no negative effect (loss of coverage, inability to make binaries, etc), going to go ahead and merge.

**Comment by budziq** [comment]
> Is this a known bug with either rust beta or skeptic?

@killercup - it was a skeptic bug due to change in how cargo beta produces Cargo.lock. It's now fixed in crates.io 0.13.2 (in a sane way not parsing Cargo.lock)

**Pull request merged**